crash after creating new accounts and navigating to the send page.

After creating new accounts and going to the send page, godcr crashes with the terminal log below. 

![account-selection-bug](https://user-images.githubusercontent.com/15804688/115117934-da39d280-9f98-11eb-95ee-b3e101659a26.gif)






panic: runtime error: index out of range [4] with length 4

goroutine 52 [running]:
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection.func2.1.1.1.1.1(0x0, 0x0, 0x18a, 0xf4240, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:686 +0x1253
gioui.org/layout.(*List).Layout(0xc0009925b0, 0x0, 0x0, 0x18a, 0xf4240, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/list.go:111 +0x165
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection.func2.1.1.1.1(0x0, 0x0, 0x18a, 0xf4215, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:667 +0x127
gioui.org/layout.Flex.Layout(0x4000001, 0x0, 0x0, 0x18a, 0xf4236, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/flex.go:99 +0xd86
github.com/planetdecred/godcr/ui.(*sendPage).popupSections(0xc000992000, 0x0, 0x0, 0x18a, 0xf4236, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:1061 +0x25b
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection.func2.1.1.1(0x0, 0x0, 0x18a, 0xf4236, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:666 +0x1a4
gioui.org/layout.Inset.Layout(0x0, 0x0, 0x141200000, 0x0, 0x0, 0x0, 0x18a, 0xf4236, 0x3f8000003f800000, 0x5071440, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection.func2.1.1(0x0, 0x0, 0x18a, 0xf4240, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:665 +0x2a6
gioui.org/layout.(*List).Layout(0xc0009922d8, 0x0, 0x0, 0x18a, 0xf4240, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/list.go:111 +0x165
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection.func2.1(0x0, 0x0, 0x18a, 0xf422c, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:653 +0xe6
gioui.org/layout.Stack.Layout(0x4bead00, 0x18a, 0x0, 0x18a, 0xf422c, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/stack.go:78 +0x6d7
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection.func2(0x18a, 0x0, 0x18a, 0xf422c, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:651 +0x2d1
gioui.org/layout.Inset.Layout(0x141200000, 0x141200000, 0x141200000, 0x141200000, 0x18a, 0x0, 0x18a, 0xf422c, 0x3f8000003f800000, 0x5071440, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1.1.1.1.1(0x19e, 0x0, 0x19e, 0xf4240, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:55 +0xc6
gioui.org/layout.(*List).Layout(0xc00098c600, 0x0, 0x0, 0x19e, 0xf4240, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/list.go:111 +0x165
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1.1.1.1(0x0, 0x0, 0x19e, 0x2c1, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:53 +0xc9
gioui.org/layout.Inset.Layout(0x141a00000, 0x141a00000, 0x141a00000, 0x141a00000, 0x0, 0x0, 0x19e, 0x2c1, 0x3f8000003f800000, 0x5071440, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1.1.1(0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:52 +0xff
gioui.org/layout.Stack.Layout(0x4bebe00, 0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout.func1.1(0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:45 +0x260
gioui.org/layout.Inset.Layout(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout.func1(0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:44 +0x133
gioui.org/layout.Stack.Layout(0x4c25100, 0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.Card.Layout(0x0, 0x0, 0x0, 0x0, 0x41600000ffffffff, 0x4160000041600000, 0xc041600000, 0x0, 0x0, 0x1c6, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/card.go:42 +0x19a
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1.1(0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:51 +0x145
gioui.org/layout.Inset.Layout(0x0, 0x43342eaa, 0x0, 0x43342eaa, 0x0, 0x0, 0x1c6, 0x2e9, 0x3f8000003f800000, 0x5071440, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/layout.go:147 +0x4e5
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2.1(0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:50 +0x101
gioui.org/layout.Direction.Layout(0x4b70408, 0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/layout.go:167 +0xce
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout.func2(0x0, 0x2e9, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:46 +0x19d
gioui.org/layout.Stack.Layout(0x4c5a900, 0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.(*Modal).Layout(0xc000979bd0, 0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/modal.go:33 +0x2ad
github.com/planetdecred/godcr/ui.(*sendPage).walletAccountSection(0xc000992000, 0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:727 +0x1c2
github.com/planetdecred/godcr/ui.(*sendPage).Layout(0xc000992000, 0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:372 +0x70a
github.com/planetdecred/godcr/ui.(*Window).SendPage.func1(0x0, 0x0, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, 0x5a13ae0, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/send_page.go:276 +0xe3
gioui.org/layout.Stack.Layout(0x4bec401, 0x32e, 0x2e9, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/pkg/mod/gioui.org@v0.0.0-20210228180843-e1248651c871/layout/stack.go:60 +0x88a
github.com/planetdecred/godcr/ui/decredmaterial.(*Theme).Background(0xc0004ac100, 0x32e, 0x2e9, 0x32e, 0x2e9, 0x3f8000003f800000, 0x5071440, 0xc0007f6490, 0xc010d192ad163c80, 0xb90eee3257, ...)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/decredmaterial/theme.go:128 +0x23d
github.com/planetdecred/godcr/ui.(*Window).Loop(0xc000442160, 0xc00011bd40)
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/ui/window.go:237 +0x61d
created by main.main
        /Users/klinsmann/go/src/github.com/raedahgroup/godcr/main.go:128 +0x657

--- a/ui/compontents.go
+++ b/ui/compontents.go
@@ -676,6 +676,22 @@ func (page *pageCommon) walletInfoPopup(gtx layout.Context) layout.Dimensions {
 	})
 }
 
+func (page pageCommon) addAccount(account walletAccount) {
+	account.evt = &gesture.Click{}
+	selectSendAccount := page.wallAcctSelector.walletAccounts.selectSendAccount
+	selectReceiveAccount := page.wallAcctSelector.walletAccounts.selectReceiveAccount
+
+	if len(selectSendAccount) > account.walletIndex {
+		account.accountIndex = len(selectSendAccount[account.walletIndex])
+		page.wallAcctSelector.walletAccounts.selectSendAccount[account.walletIndex] = append(page.wallAcctSelector.walletAccounts.selectSendAccount[account.walletIndex], account)
+	}
+
+	if len(selectReceiveAccount) > account.walletIndex {
+		account.accountIndex = len(selectReceiveAccount[account.walletIndex])
+		page.wallAcctSelector.walletAccounts.selectReceiveAccount[account.walletIndex] = append(page.wallAcctSelector.walletAccounts.selectReceiveAccount[account.walletIndex], account)
+	}
+}
+
 func (page pageCommon) handleNavEvents() {
 	for page.minimizeNavDrawerButton.Button.Clicked() {
 		*page.isNavDrawerMinimized = true

--- a/ui/wallet_page.go
+++ b/ui/wallet_page.go
@@ -15,6 +15,7 @@ import (
 	"gioui.org/widget/material"
 
 	"github.com/decred/dcrd/dcrutil"
+	"github.com/planetdecred/dcrlibwallet"
 	"github.com/planetdecred/godcr/ui/decredmaterial"
 	"github.com/planetdecred/godcr/ui/values"
 	"github.com/planetdecred/godcr/wallet"
@@ -837,12 +838,21 @@ func (pg *walletPage) Handle(common pageCommon) {
 
 		for pg.collapsibles[index].addAcctBtn.Button.Clicked() {
 			walletID := pg.walletInfo.Wallets[index].ID
+			walletIndex := index
 			go func() {
 				common.modalReceiver <- &modalLoad{
 					template: CreateAccountTemplate,
 					title:    "Create new account",
 					confirm: func(name string, passphrase string) {
-						pg.wallet.AddAccount(walletID, name, []byte(passphrase), pg.errorReceiver)
+						pg.wallet.AddAccount(walletID, name, []byte(passphrase), pg.errorReceiver, func(acct *dcrlibwallet.Account) {
+							walletAccount := walletAccount{
+								walletIndex:  walletIndex,
+								accountName:  acct.Name,
+								totalBalance: dcrutil.Amount(acct.Balance.Total).String(),
+								spendable:    dcrutil.Amount(acct.Balance.Spendable).String(),
+							}
+							common.addAccount(walletAccount)
+						})
 					},
 					confirmText: "Create",
 					cancel:      common.closeModal,

--- a/wallet/commands.go
+++ b/wallet/commands.go
@@ -99,7 +99,7 @@ func (wal *Wallet) DeleteWallet(walletID int, passphrase []byte, errChan chan er
 
 // AddAccount adds an account to a wallet.
 // It is non-blocking and sends its result or any error to wal.Send.
-func (wal *Wallet) AddAccount(walletID int, name string, pass []byte, errChan chan error) {
+func (wal *Wallet) AddAccount(walletID int, name string, pass []byte, errChan chan error, onCreate func(*dcrlibwallet.Account)) {
 	go func() {
 		var resp Response
 		wall := wal.multi.WalletWithID(walletID)
@@ -128,6 +128,22 @@ func (wal *Wallet) AddAccount(walletID int, name string, pass []byte, errChan ch
 			})
 			return
 		}
+
+		acct, err := wall.GetAccount(id)
+		if err != nil {
+			go func() {
+				errChan <- err
+			}()
+			resp.Err = err
+			wal.Send <- ResponseError(InternalWalletError{
+				Message:  "Could not fetch newly created account",
+				Affected: []int{walletID},
+				Err:      err,
+			})
+			return
+		}
+		onCreate(acct)
+
 		resp.Resp = AddedAccount{
 			ID: id,
 		}

